### PR TITLE
Pstl

### PR DESCRIPTION
**Pull request by alexstrel**
Added C++17 PSTL version. Tested on x86 (gcc-10.2 + TBB 2020.3) and nvidia march (nvc++ 2021.3)

### CODE REVIEW

**Comment by kakwok** [comment]
@alexstrel Can you add the commands you used to compile the code? I can add the support to the build script if needed

**Comment by alexstrel** [comment]
@kakwok made initial commit, not tested. pstl technology will refer to many targets, need  to adjust the script, maybe add "backend" option? BTW, made std=c++17 as a default option (as it is for the recently released  gcc-11.1.0, dpcpp and nvc++ )

File: build.py [hunk 1]
```diff
@@ -27,7 +27,11 @@
 
     "cuda":{
         "cuda":['nvcc']
-    }
+    },
+    "pstl":{
+        "pstl":['gcc','nvc++']
+    },
```
Comment (kakwok) [inline build.py:36]: @alexstrel We meant to support different backend like this: (see a few lines below for the comment on kokkos)
``` 
   "pstl":{
        "cpu":['gcc'] # add other compilers
        'cuda': ['nvc++']
    },
```

**Comment by cerati** [comment]
@kakwok @alexstrel is this PR ready to be merged?

**Comment by alexstrel** [comment]
@cerati  the branch was now checked against nvc++ (21.7)  ( both gpu and x86) as well as g++ (v. 11.2.0)  + tbb for x86. So far no issues. I need to check build script and README for the merge.

**Comment by kakwok** [comment]
@alexstrel @cerati I fixed the conflict in the build script in https://github.com/cerati/p2r-tests/pull/10/commits/594350e94967e27aa3280a0871562e8ebc3e12d4, can this be merged now?

**Comment by alexstrel** [comment]
@kakwok thanks for the update, looks good for me. (I moved dpl version into a separate feature branch since it still needs some tuning/testing before merge)

**Comment by cerati** [comment]
based on comments above it looks like all is good, so I am merging

**Pull request merged**